### PR TITLE
Add nodeos/cleos/keosd docs from develop, update README

### PR DESCRIPTION
**Pull request by lparisc**
## Change Description

- add docs folder containing product documentation from develop branch.
- update main README.md to match develop branch, keeping 2.0.x refs.

## Consensus Changes
- [ ] Consensus Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [x] Documentation Additions

### CODE REVIEW

File: docs/00_install/01_build-from-source/02_manual-build/index.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+[[info | Manual Builds are for Advanced Developers]]
+| These manual instructions are intended for advanced developers. The [Shell Scripts](../01_shell-scripts/index.md) should be the preferred method to build EOSIO from source. If the script fails or your platform is not supported, continue with the instructions below.
+
+## EOSIO Dependencies
+
+When performing a manual build, it is necessary to install specific software packages that the EOSIO software depends on. To learn more about these dependencies, visit the [EOSIO Software Dependencies](00_eosio-dependencies.md) section.
+
+## Platforms
+
+Shell commands are available to manually download, build, install, test, and uninstall the EOSIO software and dependencies for these [platforms](03_platforms/index.md).
```
Comment (arhag) [inline docs/00_install/01_build-from-source/02_manual-build/index.md]: We should avoid having a document for each variation of the platforms we support. We already discuss how to grab the dependencies with a dry-run of the script in `00_eosio-dependencies.md`. And this manual build section is really a plan C anyway (we currently recommend users to use binaries or use the script if they need source builds). 

So after the dependencies have been installed by the user, the remaining steps should be very simple and general to all the platforms we support. We should just include those directly in here. Take the "Build/Install/Test/Uinstall EOSIO" sections of the Ubuntu 18.04 unpinned instructions (which is the only platform we really need to claim we support for the manual build docs instructions at this point) and bring them in here (after making some simplifying changes which I note in the other document).

File: docs/00_install/01_build-from-source/02_manual-build/03_platforms/ubuntu-18.04-unpinned.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    cmake -DBUILD_SHARED_LIBS=OFF -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=$EOSIO_INSTALL_LOCATION .. && \
+    make -j$(nproc) && \
+    make install && \
+    rm -rf $EOSIO_INSTALL_LOCATION/mongo-cxx-driver-r3.4.0.tar.gz $EOSIO_INSTALL_LOCATION/mongo-cxx-driver-r3.4.0
+```
+
+## Build EOSIO
+These commands build the EOSIO software on the specified OS. Make sure to [Install EOSIO Dependencies](#install-eosio-dependencies) first.
+```sh
+export EOSIO_BUILD_LOCATION=$EOSIO_LOCATION/build
```
Comment (arhag) [inline docs/00_install/01_build-from-source/02_manual-build/03_platforms/ubuntu-18.04-unpinned.md]: Avoid the environment variables. Just use `~/eosio/eos/build` directly.

File: docs/00_install/01_build-from-source/02_manual-build/03_platforms/ubuntu-18.04-unpinned.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+mkdir -p $EOSIO_BUILD_LOCATION
+cd $EOSIO_BUILD_LOCATION && cmake -DCMAKE_BUILD_TYPE='Release' -DCMAKE_CXX_COMPILER='clang++-7' -DCMAKE_C_COMPILER='clang-7' -DLLVM_DIR='/usr/lib/llvm-7/lib/cmake/llvm' -DCMAKE_INSTALL_PREFIX=$EOSIO_INSTALL_LOCATION -DBUILD_MONGO_DB_PLUGIN=true ..
```
Comment (arhag) [inline docs/00_install/01_build-from-source/02_manual-build/03_platforms/ubuntu-18.04-unpinned.md]: The CMake command here should be very simple: `cmake -DCMAKE_BUILD_TYPE='Release' -DCMAKE_INSTALL_PREFIX=~/eosio/x.y/ ..`. 

Notice that the `-DBUILD_MONGO_DB_PLUGIN=true` is also removed. We have deprecated the mongo_db_plugin and we should not be telling people to build it.

File: docs/00_install/01_build-from-source/02_manual-build/03_platforms/ubuntu-18.04-unpinned.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+## Install EOSIO
+This command installs the EOSIO software on the specified OS. Make sure to [Build EOSIO](#build-eosio) first.
+```sh
+cd $EOSIO_BUILD_LOCATION && make install
+```
+
+## Test EOSIO
+These commands validate the EOSIO software installation on the specified OS. Make sure to [Install EOSIO](#install-eosio) first. (**Note**: This task is optional but recommended.)
+```sh
+$EOSIO_INSTALL_LOCATION/bin/mongod --fork --logpath $(pwd)/mongod.log --dbpath $(pwd)/mongodata
```
Comment (arhag) [inline docs/00_install/01_build-from-source/02_manual-build/03_platforms/ubuntu-18.04-unpinned.md]: With the mongo_db_plugin removed from the build (and therefore the mongo_db_plugin specific tests removed from the test suite), this line is no longer necessary.

File: docs/00_install/00_install-prebuilt-binaries.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+$ sudo yum install ./eosio-2.0.0-rc2.el7.x86_64.rpm
+```
+#### RPM Package Uninstall
+```sh
+$ sudo yum remove eosio
+```
+
+## Location of EOSIO binaries
+
+After installing the prebuilt binaries, the actual EOSIO binaries will be located under the `~/eosio/x.y/bin` folder, where `x.y` is the EOSIO release version that was installed.
```
[the remark this answers is not in the document]
Comment (lparisc) [reply docs/00_install/00_install-prebuilt-binaries.md]: fixed - [86e7ba0](https://github.com/EOSIO/eos/commit/86e7ba04126b0c3c403082ba86bbccac35a13c7f)

File: README.md [hunk 6]
```diff
@@ -20,18 +20,55 @@ Some of the groundbreaking features of EOSIO include:
 1. Designed for Parallel Execution of Context Free Validation Logic
 1. Designed for Inter Blockchain Communication
 
+## License
```
[the remark this answers is not in the document]
Comment (lparisc) [reply README.md]: moved license section down, replacing the duplicated subheading - [00e7a47](https://github.com/EOSIO/eos/commit/00e7a47aa864addb4bd77869e12bdaa95fc0b9aa)

File: README.md [hunk 7]
```diff
@@ -20,18 +20,55 @@ Some of the groundbreaking features of EOSIO include:
...
 
+## Disclaimer
+
 Block.one is neither launching nor operating any initial public blockchains based upon the EOSIO software. This release refers only to version 1.0 of our open source software. We caution those who wish to use blockchains built on EOSIO to carefully vet the companies and organizations launching blockchains based on EOSIO before disclosing any private keys to their derivative software.
 
+## Testnets
+
 There is no public testnet running currently.
 
+## Supported Operating Systems
```
[the remark this answers is not in the document]
Comment (lparisc) [reply README.md]: moved this section out of README to the shell scripts section within the docs. For one, the reader wants to know which OSes are supported when installing the software (so it should be a sub-section there); and two, the prebuilt binaries installation section already mentions the supported OSes, which is redundant - [353856e](https://github.com/EOSIO/eos/commit/353856e082be2e481a9e48379dcf368839130bfe) and [b46db79](https://github.com/EOSIO/eos/commit/b46db79a345e4134089a02b1be16c6dcbaf070f0)

File: docs/00_install/index.md [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+## Supported Operating Systems
+
+EOSIO currently supports the following operating systems:
+
+1. Amazon Linux 2
+2. CentOS 7
+3. Ubuntu 16.04
+4. Ubuntu 18.04
+5. MacOS 10.14 (Mojave)
```
[the remark this answers is not in the document]
Comment (lparisc) [reply docs/00_install/index.md]: done above - [353856e](https://github.com/EOSIO/eos/commit/353856e082be2e481a9e48379dcf368839130bfe) and [b46db79](https://github.com/EOSIO/eos/commit/b46db79a345e4134089a02b1be16c6dcbaf070f0)

File: README.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (arhag) [reply README.md]: Actually, I think it is fine to keep the "Supported Operating Systems" section duplicated in the `README.md` file, and then remove the "Prebuilt Binaries" and "Shell Scripts" sections entirely, and replace the contents of the "Software Installation" section with a short paragraph that encourages most users to use prebuilt binaries (with a link to `install-prebuilt-binaries` page of the developer website) and mentions that advanced users can build from source (with a local link to the `docs/00_install/01_build-from-source/index.md` file).

File: docs/00_install/index.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (arhag) [reply docs/00_install/index.md]: This useful note should be added here to `docs/00_install/index.md` regardless of whether or not the duplicated "Supported Operating Systems" section is removed from `README.md`.

[the remark this answers is not in the document]
Comment (lparisc) [reply docs/00_install/index.md]: done - [d5886d6](https://github.com/EOSIO/eos/commit/d5886d69ae21082a70857c2f0ab65b32b3a97fea)

File: docs/00_install/01_build-from-source/02_manual-build/index.md (hunk 1 above)
Comment (arhag) [reply docs/00_install/01_build-from-source/02_manual-build/index.md]: In the interest of time, we can hold off on this change for this PR.

File: docs/00_install/01_build-from-source/02_manual-build/03_platforms/ubuntu-18.04-unpinned.md (hunk 3 above)
Comment (arhag) [reply docs/00_install/01_build-from-source/02_manual-build/03_platforms/ubuntu-18.04-unpinned.md]: In the interest of time, we can hold off on this change for this PR.

**Pull request merged**